tests/test_input.py

```diff
@@ -114,37 +114,68 @@ def test_validate_experiment_id_2():
 
 @raises(ValueError)
 def test_validate_experiment_id_3():
-    data = {'experiment_id': 'test experiment',
-            'experiment_dir': 'data/foobar',
-            'input_features_file': 'data/feats'}
-    validate_and_populate_json_fields(data, context='rsmpredict')
+    data = {'comparison_id': 'old vs new',
+            'experiment_id_old': 'old_experiment',
+            'experiment_dir_old': 'data/old',
+            'experiment_id_new': 'new_experiment',
+            'experiment_dir_new': 'data/new',}
+    validate_and_populate_json_fields(data, context='rsmcompare')
 
 
 @raises(ValueError)
 def test_validate_experiment_id_4():
-    data = {'experiment_id_old': 'old experiment',
+    data = {'comparison_id': 'old vs new',
+            'experiment_id_old': 'old experiment',
             'experiment_dir_old': 'data/old',
-            'experiment_id_new': 'new experiment',
+            'experiment_id_new': 'new_experiment',
             'experiment_dir_new': 'data/new',}
     validate_and_populate_json_fields(data, context='rsmcompare')
 
 
 @raises(ValueError)
 def test_validate_experiment_id_5():
-    data = {'experiment_id_old': 'old_experiment',
+    data = {'experiment_id': 'this_is_a_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_really_long_id',
+            'train_file': 'data/rsmtool_smTrain.csv',
+            'test_file': 'data/rsmtool_smEval.csv',
+            'model': 'LinearRegression'}
+    validate_and_populate_json_fields(data)
+
+
+@raises(ValueError)
+def test_validate_experiment_id_6():
+    data = {'experiment_id': 'this is a really really really really really really really really really really really really really really really really really really really really really really really really really really really long id',
```

Comment:
It's basically that even though it has spaces, it will fail because of length first because of how the exception raising is ordered in `check_id_fields()`.